server/src/Packages/StripeBeforeCheckout.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Gernzy\Server\Packages;
+
+use Gernzy\Server\Classes\ActionClass;
+use Gernzy\Server\Services\ActionInterface;
+use Illuminate\Support\Str;
+
+class StripeBeforeCheckout implements ActionInterface
+{
+    public function __construct()
+    {
+    }
+
+    public function run(ActionClass $action)
+    {
+        $data = $action->getOriginalData();
+
+        // Add some third party specific data
+        array_push($data, [
+            'coupon' => Str::random(12),
+            'date' => date("Y-m-d H:i:s")
+        ]);
+
+        $action->attachData(StripeBeforeCheckout::class, $data);
```

Comment:
@AlexeiaW one thing I'm curious about maybe we can investigate before merge. Is there some way we can force users to only pass in classes that implement ActionInterface. If there's some check we can do in the event service or similar to throw an error if they pass in a class that isn't implementing that. It just creates a stricter set of rules to follow, or they could pass in an arbitrary string.